pkg/cmd/kubeblocks/upgrade.go

```diff
@@ -92,6 +93,15 @@ func newUpgradeCmd(f cmdutil.Factory, streams genericiooptions.IOStreams) *cobra
 
 func (o *InstallOptions) Upgrade() error {
 	klog.V(1).Info("##### Start to upgrade KubeBlocks #####")
+	// check helm release status
+	status, err := helm.GetHelmReleaseStatus(o.HelmCfg, nil, types.KubeBlocksChartName)
+	if err != nil {
+		return fmt.Errorf("failed to get Helm release status: %v", err)
+	}
+	if status == release.StatusFailed {
```

Comment:
Please reconfirm which states would prevent the helm upgrade, and make sure to check all related states in advance.

Reply:
Only releases with a deployed status are supported for upgrade. Releases in other statuses—unknown, uninstalled, superseded, failed, uninstalling, pending-install, pending-upgrade, or pending-rollback—are not allowed.